From b0eefa018755010f5d6a44fdceaa8a9fb0b5c242 Mon Sep 17 00:00:00 2001
From: drisspg <drisspguessous@gmail.com>
Date: Thu, 1 Feb 2024 17:02:25 -0800
Subject: [PATCH] Add a missing platform include

---
 include/cutlass/bfloat16.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/cutlass/bfloat16.h b/include/cutlass/bfloat16.h
index 92d4bf92a..75cadbfa4 100644
--- a/include/cutlass/bfloat16.h
+++ b/include/cutlass/bfloat16.h
@@ -57,6 +57,7 @@
 
 #include <cuda_bf16.h>
 #include "cutlass/cutlass.h"
+#include "cutlass/platform/platform.h"
 
 namespace cutlass {
 
