diff --git a/aten/src/ATen/native/quantized/cpu/qnnpack/CMakeLists.txt b/aten/src/ATen/native/quantized/cpu/qnnpack/CMakeLists.txt
index fd6b7ff551d..8206a671dc6 100644
--- a/aten/src/ATen/native/quantized/cpu/qnnpack/CMakeLists.txt
+++ b/aten/src/ATen/native/quantized/cpu/qnnpack/CMakeLists.txt
@@ -325,13 +325,16 @@ set_target_properties(pytorch_qnnpack PROPERTIES PUBLIC_HEADER include/qnnpack_f
 
 # ---[ Configure clog
 if(NOT TARGET clog)
-  set(CLOG_BUILD_TESTS OFF CACHE BOOL "")
-  set(CLOG_RUNTIME_TYPE "${CPUINFO_RUNTIME_TYPE}" CACHE STRING "")
-  add_subdirectory(
-    "${CLOG_SOURCE_DIR}"
-    "${CONFU_DEPENDENCIES_BINARY_DIR}/clog")
-  # We build static version of clog but a dynamic library may indirectly depend on it
-  set_property(TARGET clog PROPERTY POSITION_INDEPENDENT_CODE ON)
+  find_path(CLOG_INCLUDE_DIR clog.h PATHS /usr/include)
+  find_library(CLOG_LIBRARY clog PATHS /usr/@LIBDIR@)
+  if (NOT CLOG_LIBRARY)
+    message(FATAL_ERROR "Cannot find clog")
+  endif()
+  message("Found clog: ${CLOG_LIBRARY}")
+  add_library(clog INTERFACE IMPORTED)
+  set_target_properties(clog PROPERTIES
+    INTERFACE_INCLUDE_DIRECTORIES "${CLOG_INCLUDE_DIR}"
+    INTERFACE_LINK_LIBRARIES "${CLOG_LIBRARY}")
 endif()
 target_link_libraries(pytorch_qnnpack PUBLIC clog)
 
