diff --git a/functorch/CMakeLists.txt b/functorch/CMakeLists.txt
index 3e57807..9fe4930 100644
--- a/functorch/CMakeLists.txt
+++ b/functorch/CMakeLists.txt
@@ -35,4 +35,4 @@ endif()
 if(NOT ${TORCH_PYTHON_LINK_FLAGS} STREQUAL "")
   set_target_properties(${PROJECT_NAME} PROPERTIES LINK_FLAGS ${TORCH_PYTHON_LINK_FLAGS})
 endif()
-install(TARGETS ${PROJECT_NAME} DESTINATION "${CMAKE_CURRENT_SOURCE_DIR}")
+install(TARGETS ${PROJECT_NAME} DESTINATION "${TORCH_INSTALL_LIB_DIR}")
