diff --git a/cmake/Modules/FindBLIS.cmake b/cmake/Modules/FindBLIS.cmake
index 3be90c19ea1..ddf143ada02 100644
--- a/cmake/Modules/FindBLIS.cmake
+++ b/cmake/Modules/FindBLIS.cmake
@@ -21,6 +21,7 @@ SET(BLIS_INCLUDE_SEARCH_PATHS
 )
 
 SET(BLIS_LIB_SEARCH_PATHS
+  /usr/@LIBDIR@
   /lib/blis
   /lib64/blis
   /usr/lib/blis
@@ -31,7 +32,7 @@ SET(BLIS_LIB_SEARCH_PATHS
   $ENV{BLIS_HOME}/lib
 )
 
-FIND_PATH(BLIS_INCLUDE_DIR NAMES cblas.h blis.h
+FIND_PATH(BLIS_INCLUDE_DIR NAMES blis.h cblas.h
           PATHS ${BLIS_INCLUDE_SEARCH_PATHS})
 #    Check include files
 IF(NOT BLIS_INCLUDE_DIR)
@@ -42,6 +43,9 @@ ENDIF()
 
 
 FIND_LIBRARY(BLIS_LIB NAMES blis PATHS ${BLIS_LIB_SEARCH_PATHS})
+IF(NOT BLIS_LIB)
+        FIND_LIBRARY(BLIS_LIB NAMES blis-mt PATHS ${BLIS_LIB_SEARCH_PATHS})
+ENDIF()
 #    Check libraries
 IF(NOT BLIS_LIB)
         SET(BLIS_FOUND OFF)
