diff -ur torch-fidelity-3dcb32c877dbbcbdd5023806a69c000e19fd0285_orig/setup.py torch-fidelity-3dcb32c877dbbcbdd5023806a69c000e19fd0285/setup.py
--- torch-fidelity-3dcb32c877dbbcbdd5023806a69c000e19fd0285_orig/setup.py	2023-12-24 13:30:16.414898076 +0800
+++ torch-fidelity-3dcb32c877dbbcbdd5023806a69c000e19fd0285/setup.py	2023-12-24 13:30:49.194896883 +0800
@@ -38,7 +38,7 @@
     long_description_content_type='text/plain',
     install_requires=requirements,
     python_requires='>=3.6',
-    packages=find_packages(),
+    packages=find_packages(exclude=["tests", "tests.*"]),
     author='Anton Obukhov',
     license='Apache License 2.0',
     url='https://www.github.com/toshas/torch-fidelity',
@@ -52,4 +52,4 @@
             'fidelity=torch_fidelity.fidelity:main',
         ],
     },
-)
\ No newline at end of file
+)
