diff -ur grpcio-1.62.1/Makefile grpcio-1.62.1_modified/Makefile
--- grpcio-1.62.1/Makefile	2024-03-08 08:15:56.000000000 +0800
+++ grpcio-1.62.1_modified/Makefile	2024-03-17 19:52:17.001366810 +0800
@@ -322,7 +322,7 @@
 HOST_LDXX ?= $(LDXX)
 
 CFLAGS += -std=c11 $(W_EXTRA_SEMI)
-CXXFLAGS += -std=c++14
+CXXFLAGS += -std=c++17
 ifeq ($(SYSTEM),Darwin)
 CXXFLAGS += -stdlib=libc++
 LDFLAGS += -framework CoreFoundation
diff -ur grpcio-1.62.1/setup.py grpcio-1.62.1_modified/setup.py
--- grpcio-1.62.1/setup.py	2024-03-08 08:15:56.000000000 +0800
+++ grpcio-1.62.1_modified/setup.py	2024-03-17 19:51:55.111367036 +0800
@@ -215,7 +215,7 @@
     )
     cxx = shlex.split(os.environ.get("CXX", "c++"))
     cpp_test = subprocess.Popen(
-        cxx + ["-x", "c++", "-std=c++14", "-"],
+        cxx + ["-x", "c++", "-std=c++17", "-"],
         stdin=PIPE,
         stdout=PIPE,
         stderr=PIPE,
@@ -226,7 +226,7 @@
     # Double-check to see if -latomic actually can solve the problem.
     # https://github.com/grpc/grpc/issues/22491
     cpp_test = subprocess.Popen(
-        cxx + ["-x", "c++", "-std=c++14", "-", "-latomic"],
+        cxx + ["-x", "c++", "-std=c++17", "-", "-latomic"],
         stdin=PIPE,
         stdout=PIPE,
         stderr=PIPE,
@@ -254,7 +254,7 @@
         EXTRA_ENV_COMPILE_ARGS += " /MT"
     elif "linux" in sys.platform:
         # GCC by defaults uses C17 so only C++14 needs to be specified.
-        EXTRA_ENV_COMPILE_ARGS += " -std=c++14"
+        EXTRA_ENV_COMPILE_ARGS += " -std=c++17"
         EXTRA_ENV_COMPILE_ARGS += (
             " -fvisibility=hidden -fno-wrapv -fno-exceptions"
         )
diff -ur grpcio-1.62.1/src/python/grpcio/commands.py grpcio-1.62.1_modified/src/python/grpcio/commands.py
--- grpcio-1.62.1/src/python/grpcio/commands.py	2024-03-08 08:15:56.000000000 +0800
+++ grpcio-1.62.1_modified/src/python/grpcio/commands.py	2024-03-17 19:52:38.661366486 +0800
@@ -256,7 +256,7 @@
                 # TODO(lidiz) Remove the generated a.out for success tests.
                 cc = os.environ.get("CC", "cc")
                 cc_test = subprocess.Popen(
-                    [cc, "-x", "c", "-std=c++14", "-"],
+                    [cc, "-x", "c", "-std=c++17", "-"],
                     stdin=subprocess.PIPE,
                     stdout=subprocess.PIPE,
                     stderr=subprocess.PIPE,
diff -ur grpcio-1.62.1/src/python/grpcio/grpc/_cython/cygrpc.cpp grpcio-1.62.1_modified/src/python/grpcio/grpc/_cython/cygrpc.cpp
--- grpcio-1.62.1/src/python/grpcio/grpc/_cython/cygrpc.cpp	2024-03-08 08:16:14.000000000 +0800
+++ grpcio-1.62.1_modified/src/python/grpcio/grpc/_cython/cygrpc.cpp	2024-03-17 19:52:54.181366396 +0800
@@ -60,7 +60,7 @@
             "src/core/lib/surface/call.h"
         ],
         "extra_compile_args": [
-            "-std=c++14",
+            "-std=c++17",
             "-fvisibility=hidden",
             "-fno-wrapv",
             "-fno-exceptions"
