diff --git a/fbgemm_gpu/setup.py b/fbgemm_gpu/setup.py
index 1a809e27..012f48be 100644
--- a/fbgemm_gpu/setup.py
+++ b/fbgemm_gpu/setup.py
@@ -17,7 +17,6 @@ import textwrap
 from datetime import date
 from typing import List, Optional
 
-import setuptools_git_versioning as gitversion
 import torch
 from setuptools.command.install import install as PipInstall
 from skbuild import setup
@@ -39,14 +38,8 @@ def generate_package_version(package_name: str, version_variant: str):
         version = (f"0.0.{random.randint(0, 1000)}",)
 
     else:
-        # Use git tag / branch / commit info to generate a PEP-440-compliant version string
-        print("[SETUP.PY] Package is for RELEASE: using git info for the versioning")
-        print(
-            f"[SETUP.PY] TAG: {gitversion.get_tag()}, BRANCH: {gitversion.get_branch()}, SHA: {gitversion.get_sha()}"
-        )
-        # Remove the local version identifier, if any (e.g. 0.4.0rc0.post0+git.6a63116c.dirty => 0.4.0rc0.post0)
-        # Then remove post0 (keep postN for N > 0) (e.g. 0.4.0rc0.post0 => 0.4.0rc0)
-        version = re.sub(".post0$", "", gitversion.version_from_git().split("+")[0])
+        print("[SETUP.PY] Package is for RELEASE")
+        version = '0.5.0'
     version = str(version) + version_variant
     print(f"[SETUP.PY] Setting the package version: {version}")
     return version
