From e9a3213524a0abd609ac7330cf170b9e19917d39 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C8=98tefan=20Talpalaru?= <stefantalpalaru@yahoo.com>
Date: Wed, 23 Oct 2024 12:36:26 +0200
Subject: [PATCH] fix CUDA build with NVCC_FLAGS in env (#8692)

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 05a07c826f..c2be57e977 100644
--- a/setup.py
+++ b/setup.py
@@ -127,7 +127,7 @@ def get_macros_and_flags():
             if NVCC_FLAGS is None:
                 nvcc_flags = []
             else:
-                nvcc_flags = nvcc_flags.split(" ")
+                nvcc_flags = NVCC_FLAGS.split(" ")
         extra_compile_args["nvcc"] = nvcc_flags
 
     if sys.platform == "win32":
-- 
2.45.2

