diff --git a/WORKSPACE b/WORKSPACE
index 539c37b12d1..a2a6f86aeb2 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -23,6 +23,23 @@ load("@//tensorflow:workspace3.bzl", "tf_workspace3")
 
 tf_workspace3()
 
+http_archive(
+    name = "rules_cc",
+    sha256 = "d9bdd3ec66b6871456ec9c965809f43a0901e692d754885e89293807762d3d80",
+    strip_prefix = "rules_cc-0.0.13",
+    urls = ["https://github.com/bazelbuild/rules_cc/releases/download/0.0.13/rules_cc-0.0.13.tar.gz"],
+)
+
+http_archive(
+    name = "rules_python",
+    sha256 = "778aaeab3e6cfd56d681c89f5c10d7ad6bf8d2f1a72de9de55b23081b2d31618",
+    strip_prefix = "rules_python-0.34.0",
+    urls = ["https://github.com/bazelbuild/rules_python/releases/download/0.34.0/rules_python-0.34.0.tar.gz"],
+)
+
+load("@rules_python//python:repositories.bzl", "py_repositories")
+py_repositories()
+
 load("@//tensorflow:workspace2.bzl", "tf_workspace2")
 
 tf_workspace2()
diff --git a/tensorflow/workspace2.bzl b/tensorflow/workspace2.bzl
index be83c971749..e644e005f07 100644
--- a/tensorflow/workspace2.bzl
+++ b/tensorflow/workspace2.bzl
@@ -381,6 +381,13 @@ def _tf_repositories():
         urls = tf_mirror_urls("https://github.com/abseil/abseil-py/archive/refs/tags/v1.0.0.tar.gz"),
     )
 
+    tf_http_archive(
+        name = "protobuf",
+        sha256 = "5be917c830f7504976d775a9fbadf574746fa17eb753cb6e2a2e7039bf7b392c",
+        strip_prefix = "protobuf-3.27.4",
+        urls = tf_mirror_urls("https://github.com/protocolbuffers/protobuf/archive/v3.27.4.zip"),
+    )
+
     tf_http_archive(
         name = "com_google_protobuf",
         patch_file = ["//third_party/protobuf:protobuf.patch"],
