From d79401c1e3a3ccf7fb3aa659e0591e21e7a03d49 Mon Sep 17 00:00:00 2001
From: wangjiezhe <wangjiezhe@gmail.com>
Date: Thu, 21 Dec 2023 22:22:35 +0800
Subject: [PATCH 10/17] systemlib: fix missing `:osx` in pybind11

---
 third_party/systemlibs/pybind11.BUILD | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/third_party/systemlibs/pybind11.BUILD b/third_party/systemlibs/pybind11.BUILD
index 44e7496fc25..41d054c84ce 100644
--- a/third_party/systemlibs/pybind11.BUILD
+++ b/third_party/systemlibs/pybind11.BUILD
@@ -6,3 +6,8 @@ cc_library(
         "@local_tsl//third_party/python_runtime:headers",
     ],
 )
+
+config_setting(
+    name = "osx",
+    constraint_values = ["@platforms//os:osx"],
+)
-- 
2.49.0

