diff -ur wandb-0.17.0/core/hatch.py wandb-0.17.0_modified/core/hatch.py
--- wandb-0.17.0/core/hatch.py	2024-05-08 05:48:18.000000000 +0800
+++ wandb-0.17.0_modified/core/hatch.py	2024-05-14 16:41:43.641280764 +0800
@@ -62,12 +62,12 @@
         ext_ld_flags = " ".join(
             [
                 # Use https://en.wikipedia.org/wiki/Gold_(linker)
-                "-fuse-ld=gold",
+                "-fuse-ld=bfd",
                 # Set the --weak-unresolved-symbols option in gold, converting
                 # unresolved symbols to weak references. This is necessary to
                 # build a Go binary with cgo on Linux, where the NVML libraries
                 # needed for Nvidia GPU monitoring may not be available at build time.
-                "-Wl,--weak-unresolved-symbols",
+                "-Wl,-z dynamic-undefined-weak",
             ]
         )
         flags += ["-extldflags", f'"{ext_ld_flags}"']
