diff -ur grpcio-tools-1.62.1/grpc_tools/_protoc_compiler.cpp grpcio-tools-1.62.1_modified/grpc_tools/_protoc_compiler.cpp
--- grpcio-tools-1.62.1/grpc_tools/_protoc_compiler.cpp	2024-03-08 08:24:47.000000000 +0800
+++ grpcio-tools-1.62.1_modified/grpc_tools/_protoc_compiler.cpp	2024-03-17 19:57:57.241356790 +0800
@@ -13,7 +13,7 @@
             "grpc_tools/main.h"
         ],
         "extra_compile_args": [
-            "-std=c++14",
+            "-std=c++17",
             "-fno-wrapv",
             "-frtti"
         ],
diff -ur grpcio-tools-1.62.1/setup.py grpcio-tools-1.62.1_modified/setup.py
--- grpcio-tools-1.62.1/setup.py	2024-03-08 08:15:57.000000000 +0800
+++ grpcio-tools-1.62.1_modified/setup.py	2024-03-17 19:57:45.181357940 +0800
@@ -90,7 +90,7 @@
     )
     cxx = os.environ.get("CXX", "c++")
     cpp_test = subprocess.Popen(
-        [cxx, "-x", "c++", "-std=c++14", "-"],
+        [cxx, "-x", "c++", "-std=c++17", "-"],
         stdin=PIPE,
         stdout=PIPE,
         stderr=PIPE,
@@ -101,7 +101,7 @@
     # Double-check to see if -latomic actually can solve the problem.
     # https://github.com/grpc/grpc/issues/22491
     cpp_test = subprocess.Popen(
-        [cxx, "-x", "c++", "-std=c++14", "-", "-latomic"],
+        [cxx, "-x", "c++", "-std=c++17", "-", "-latomic"],
         stdin=PIPE,
         stdout=PIPE,
         stderr=PIPE,
@@ -147,7 +147,7 @@
         EXTRA_ENV_COMPILE_ARGS += " /MT"
     elif "linux" in sys.platform or "darwin" in sys.platform:
         # GCC & Clang by defaults uses C17 so only C++14 needs to be specified.
-        EXTRA_ENV_COMPILE_ARGS += " -std=c++14"
+        EXTRA_ENV_COMPILE_ARGS += " -std=c++17"
         EXTRA_ENV_COMPILE_ARGS += " -fno-wrapv -frtti"
 if EXTRA_ENV_LINK_ARGS is None:
     EXTRA_ENV_LINK_ARGS = ""
