diff -ur cutlass-3.3.0_orig/setup.cfg cutlass-3.3.0/setup.cfg
--- cutlass-3.3.0_orig/setup.cfg	2024-01-26 01:09:34.181892505 +0800
+++ cutlass-3.3.0/setup.cfg	2024-01-26 01:10:17.841884081 +0800
@@ -17,17 +17,9 @@
   cutlass.op
   cutlass.utils
   cutlass_library
-  cutlass_library.source
   pycute
 package_dir =
   cutlass=python/cutlass
   cutlass_library=python/cutlass_library
-  cutlass_library.source=.
   pycute=python/pycute
 include_package_data = True
-
-[options.package_data]
-cutlass_library.source = include/**/*, examples/**/*, tools/**/*
-
-[options.exclude_package_data]
-cutlass_library.source = include/**/*.py, examples/**/*.py, tools/**/*.py
