diff -ur magma-2.7.2_orig/CMakeLists.txt magma-2.7.2/CMakeLists.txt
--- magma-2.7.2_orig/CMakeLists.txt	2023-08-25 21:15:37.000000000 +0800
+++ magma-2.7.2/CMakeLists.txt	2023-12-21 17:42:08.979201999 +0800
@@ -117,7 +117,7 @@
 
   # ----------------------------------------
   # locate CUDA libraries
-  set( GPU_TARGET "Kepler Maxwell Pascal" CACHE STRING "CUDA architectures to compile for; one or more of Fermi, Kepler, Maxwell, Pascal, Volta, Turing, Ampere, Hopper, or valid sm_[0-9][0-9]" )
+  set( GPU_TARGET "Kepler Maxwell Pascal" CACHE STRING "CUDA architectures to compile for; one or more of Fermi, Kepler, Maxwell, Pascal, Volta, Turing, Ampere, Lovelace, Hopper, or valid sm_[0-9][0-9]" )
   find_package( CUDAToolkit )
   if (CUDAToolkit_FOUND)
     message( STATUS "Found CUDA ${CUDA_VERSION}" )
@@ -168,6 +168,10 @@
         set( GPU_TARGET "${GPU_TARGET} sm_80" )
     endif()
 
+    if (GPU_TARGET MATCHES Lovelace)
+        set( GPU_TARGET "${GPU_TARGET} sm_89" )
+    endif()
+
     if (GPU_TARGET MATCHES Hopper)
         set( GPU_TARGET "${GPU_TARGET} sm_90" )
     endif()
@@ -284,6 +288,14 @@
         message( STATUS "    compile for CUDA arch 8.0 (Ampere)" )
     endif()
 
+    if (GPU_TARGET MATCHES sm_89)
+        if (NOT MIN_ARCH)
+            set( MIN_ARCH 890 )
+        endif()
+        list(APPEND __cuda_architectures 89)
+        message( STATUS "    compile for CUDA arch 8.9 (Ada Lovelace)" )
+    endif()
+
     if (GPU_TARGET MATCHES sm_90)
         if (NOT MIN_ARCH)
             set( MIN_ARCH 900 )
@@ -293,7 +305,7 @@
     endif()
 
     if (NOT MIN_ARCH)
-        message( FATAL_ERROR "GPU_TARGET must contain one or more of Fermi, Kepler, Maxwell, Pascal, Volta, Turing, Ampere, or valid sm_[0-9][0-9]" )
+        message( FATAL_ERROR "GPU_TARGET must contain one or more of Fermi, Kepler, Maxwell, Pascal, Volta, Turing, Ampere, Lovelace, Hopper or valid sm_[0-9][0-9]" )
     endif()
 
     set(CUDA_ARCHITECTURES "${__cuda_architectures}")
diff -ur magma-2.7.2_orig/Makefile magma-2.7.2/Makefile
--- magma-2.7.2_orig/Makefile	2023-08-25 21:20:12.000000000 +0800
+++ magma-2.7.2/Makefile	2023-12-21 16:45:45.309593160 +0800
@@ -152,6 +152,9 @@
 	ifneq ($(findstring Ampere, $(GPU_TARGET)),)
 		CUDA_ARCH_ += sm_80
 	endif
+	ifneq ($(findstring Lovelace, $(GPU_TARGET)),)
+		CUDA_ARCH_ += sm_89
+	endif
 	ifneq ($(findstring Hopper, $(GPU_TARGET)),)
 		CUDA_ARCH_ += sm_90
 	endif
@@ -169,7 +172,7 @@
 	# See also $(info compile for ...) in Makefile
 
 
-    CUDA_ARCH_UNKNOWN_ = $(filter-out sm_% Kepler Maxwell Pascal Volta Turing Ampere Hopper, $(CUDA_ARCH_))
+    CUDA_ARCH_UNKNOWN_ = $(filter-out sm_% Kepler Maxwell Pascal Volta Turing Ampere Lovelace Hopper, $(CUDA_ARCH_))
     ifneq ($(CUDA_ARCH_UNKNOWN_),)
         $(error ERROR: unknown `$(CUDA_ARCH_UNKNOWN_)` in GPU_TARGET)
     endif
@@ -189,7 +192,7 @@
 
 	# Check for empty
     ifeq ($(NV_SM),)
-        $(error ERROR: unknown `GPU_TARGET=$(GPU_TARGET)`. Set cuda_arch to one or more of Kepler, Maxwell, Pascal, Volta, Turing, Ampere, Hopper, or valid sm_XX from nvcc -h)
+        $(error ERROR: unknown `GPU_TARGET=$(GPU_TARGET)`. Set cuda_arch to one or more of Kepler, Maxwell, Pascal, Volta, Turing, Ampere, Lovelace, Hopper, or valid sm_XX from nvcc -h)
     else
         # Get last option (last 2 words) of nv_compute.
         nwords := $(words $(NV_COMP))
@@ -205,7 +208,7 @@
 	# (add zero, so its comparable to '__CUDA_ARCH__')
     CUDA_ARCH_MIN := $(word 1, $(SMS))0
     ifeq ($(CUDA_ARCH_MIN),)
-        $(error GPU_TARGET, currently $(GPU_TARGET), must contain one or more of Fermi, Kepler, Maxwell, Pascal, Volta, Turing, Ampere, Hopper, or valid sm_[0-9][0-9]. Please edit your make.inc file)
+        $(error GPU_TARGET, currently $(GPU_TARGET), must contain one or more of Fermi, Kepler, Maxwell, Pascal, Volta, Turing, Ampere, Lovelace, Hopper, or valid sm_[0-9][0-9]. Please edit your make.inc file)
     endif
 else ifeq ($(BACKEND),hip)
 
