diff --git a/torch/__init__.py b/torch/__init__.py
index 4f37bb42be8..9e465c25271 100644
--- a/torch/__init__.py
+++ b/torch/__init__.py
@@ -1944,7 +1944,7 @@ from torch.serialization import load, save
 def _manager_path():
     if _running_with_deploy() or platform.system() == "Windows":
         return b""
-    path = get_file_path("torch", "bin", "torch_shm_manager")
+    path = "/usr/bin/torch_shm_manager"
     prepare_multiprocessing_environment(get_file_path("torch"))
     if not os.path.exists(path):
         raise RuntimeError("Unable to find torch_shm_manager at " + path)
