diff --git a/pyproject.toml b/pyproject.toml
index ebf92d9..7086f7f 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -14,8 +14,8 @@ classifiers = [
 ]
 
 [tool.setuptools]
-packages = ["cudnn", "include", "cudnn.gemm_swiglu", "cudnn.gemm_amax"]
-package-dir = {"" = "python", "include" = "include"}
+packages = ["cudnn", "cudnn.gemm_swiglu", "cudnn.gemm_amax"]
+package-dir = {"" = "python"}
 include-package-data = true
 
 [project.urls]
