diff --git a/test/python/test_matmul_bias_relu.py b/test/python/test_matmul_bias_relu.py
index 2bc0007..3f66456 100644
--- a/test/python/test_matmul_bias_relu.py
+++ b/test/python/test_matmul_bias_relu.py
@@ -33,7 +33,7 @@ def get_cc():
     LooseVersion(cudnn.backend_version_string()) < "8.9.6",
     reason="requires cudnn 8.9.6 or higher",
 )
-@pytest.mark.skipif(torch.cuda.get_device_capability()[0] < 9, reason="requires Hopper or newer arch")
+@pytest.mark.skipif(True, reason="requires Hopper or newer arch")
 @pytest.mark.L0
 @torch_fork_set_rng(seed=0)
 def test_int8_bf16_matmul(cudnn_handle):
@@ -97,7 +97,7 @@ MMA_data_type_options = [torch.bfloat16, torch.float16, torch.float32]
     LooseVersion(cudnn.backend_version_string()) < "8.9.6",
     reason="requires cudnn 8.9.6 or higher",
 )
-@pytest.mark.skipif(torch.cuda.get_device_capability()[0] < 9, reason="requires Hopper or newer arch")
+@pytest.mark.skipif(True, reason="requires Hopper or newer arch")
 @pytest.mark.parametrize("A_data_type", A_data_type_options)
 @pytest.mark.parametrize("B_data_type", B_data_type_options)
 @pytest.mark.parametrize("MMA_data_type", MMA_data_type_options)
diff --git a/test/python/test_slice.py b/test/python/test_slice.py
index 745d91e..a462324 100644
--- a/test/python/test_slice.py
+++ b/test/python/test_slice.py
@@ -7,7 +7,7 @@ from looseversion import LooseVersion
 from test_utils import torch_fork_set_rng
 
 
-@pytest.mark.skipif(torch.cuda.get_device_capability()[0] < 9, reason="requires Hopper or newer arch")
+@pytest.mark.skipif(True, reason="requires Hopper or newer arch")
 @pytest.mark.L0
 @torch_fork_set_rng(seed=0)
 def test_int8_bf16_matmul_slice(cudnn_handle):
