diff --git a/tests/unit_tests/CMakeLists.txt b/tests/unit_tests/CMakeLists.txt
index 96a8539..0611a8e 100644
--- a/tests/unit_tests/CMakeLists.txt
+++ b/tests/unit_tests/CMakeLists.txt
@@ -310,42 +310,6 @@ add_subdirectory(data)
 
 set(TESTS_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/unit_tests")
 
-if(NOT BUILD_SPARSE_TESTS_OUTOFSRCTREE)
-    install(FILES CMakeLists.txt DESTINATION ${TESTS_INSTALL_DIR})
-    install(FILES common_data_utils.cpp DESTINATION ${TESTS_INSTALL_DIR})
-    install(FILES common_data_utils.h DESTINATION ${TESTS_INSTALL_DIR}/include)
-    install(FILES createcsc_ut_functions.hpp DESTINATION ${TESTS_INSTALL_DIR}/include)
-    install(FILES cg_ut_functions.hpp DESTINATION ${TESTS_INSTALL_DIR}/include)
-    install(FILES gmres_ut_functions.hpp DESTINATION ${TESTS_INSTALL_DIR}/include)
-    install(FILES data/CMakeLists.txt DESTINATION ${TESTS_INSTALL_DIR}/data)
-    install(FILES data/tinyc.mtx DESTINATION ${TESTS_INSTALL_DIR}/data)
-    install(FILES data/tinyr.mtx DESTINATION ${TESTS_INSTALL_DIR}/data)
-    install(FILES ../common/aoclsparse_template_specialization.cpp DESTINATION ${TESTS_INSTALL_DIR}/common)
-    install(FILES ../common/aoclsparse_utility.cpp DESTINATION ${TESTS_INSTALL_DIR}/common)
-    install(FILES ../include/aoclsparse_interface.hpp DESTINATION ${TESTS_INSTALL_DIR}/include)
-    install(FILES ../include/aoclsparse_reference.hpp DESTINATION ${TESTS_INSTALL_DIR}/include)
-    install(FILES ../include/aoclsparse_random.hpp DESTINATION ${TESTS_INSTALL_DIR}/include)
-    install(FILES ../include/aoclsparse_arguments.hpp DESTINATION ${TESTS_INSTALL_DIR}/include)
-    install(FILES ../include/aoclsparse_datatype2string.hpp DESTINATION ${TESTS_INSTALL_DIR}/include)
-    install(FILES ../include/aoclsparse_init.hpp DESTINATION ${TESTS_INSTALL_DIR}/include)
-    install(FILES ../include/aoclsparse_test.hpp DESTINATION ${TESTS_INSTALL_DIR}/include)
-    install(FILES ../include/aoclsparse_utility.hpp DESTINATION ${TESTS_INSTALL_DIR}/include)
-    install(FILES ../include/aoclsparse_check.hpp DESTINATION ${TESTS_INSTALL_DIR}/include)
-    install(FILES ../../library/src/include/aoclsparse_descr.h DESTINATION ${TESTS_INSTALL_DIR}/include)
-    install(FILES ../../library/src/include/aoclsparse_mat_structures.hpp DESTINATION ${TESTS_INSTALL_DIR}/include)
-    install(FILES ../../library/src/include/aoclsparse_mtx_dispatcher.hpp DESTINATION ${TESTS_INSTALL_DIR}/include)
-    install(FILES ../../library/src/extra/aoclsparse_utils.hpp DESTINATION ${TESTS_INSTALL_DIR}/include)
-    install(FILES ${CMAKE_SOURCE_DIR}/cmake/Dependencies.cmake DESTINATION ${TESTS_INSTALL_DIR})
-    install(FILES ${CMAKE_SOURCE_DIR}/cmake/Coverage.cmake DESTINATION ${TESTS_INSTALL_DIR})
-    install(FILES ../benchmarks/bench_tests.cmake DESTINATION ${TESTS_INSTALL_DIR})
-    #--------------------------------
-    # Install KT kernel source files
-    #--------------------------------
-    install(FILES kt_kernels.cpp DESTINATION ${TESTS_INSTALL_DIR})
-    install(FILES ktlvl2_kernels.cpp DESTINATION ${TESTS_INSTALL_DIR})
-    #--------------------------------
-endif()
-
 # -----------------------------------------------------------------------
 # Build KT test dependencies
 # -----------------------------------------------------------------------
@@ -396,9 +360,6 @@ foreach(TNAME ${GTEST_EXECUTABLES})
     target_compile_definitions(${TNAME} PUBLIC ${AOCLSPARSE_ILP64})
 
     target_compile_options(${TNAME} PUBLIC $<$<CONFIG:Debug>:${COMPILER_FLAGS_DEBUG}> "-mno-avx512f;")
-    if(NOT BUILD_SPARSE_TESTS_OUTOFSRCTREE)
-        install(FILES ${TNAME}.cpp DESTINATION ${TESTS_INSTALL_DIR})
-    endif()
 endforeach()
 
 
