diff --git a/tests/unit_tests/CMakeLists.txt b/tests/unit_tests/CMakeLists.txt
index b5c7615..9fe3b6f 100644
--- a/tests/unit_tests/CMakeLists.txt
+++ b/tests/unit_tests/CMakeLists.txt
@@ -144,30 +144,7 @@ else (BUILD_SPARSE_TESTS_OUTOFSRCTREE)
 endif (BUILD_SPARSE_TESTS_OUTOFSRCTREE)
 message(STATUS "  \$SPARSE_LIBRARY......${SPARSE_LIBRARY}")
 
-include(FetchContent)
-set(FETCHCONTENT_QUIET off)
-
-# by default, the third party sources (google test) are located inside <build_directory>/external.
-if(NOT CMAKE_EXTERNAL_BUILD_DIR)
-    set(CMAKE_EXTERNAL_BUILD_DIR "./external")
-endif()
-get_filename_component(fc_base ${CMAKE_EXTERNAL_BUILD_DIR}
-                       REALPATH BASE_DIR "${CMAKE_BINARY_DIR}")
-set(FETCHCONTENT_BASE_DIR ${fc_base})
-FetchContent_Declare(
-  googletest
-  GIT_REPOSITORY https://github.com/google/googletest.git
-  GIT_TAG v1.15.2
-)
-# For Windows: Prevent overriding the parent project's compiler/linker settings
-set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
-
-FetchContent_GetProperties(googletest)
-if(NOT googletest_POPULATED)
-  FetchContent_Populate(googletest)
-  # Avoid installing gtest libs/headers by default
-  add_subdirectory(${googletest_SOURCE_DIR} ${googletest_BINARY_DIR} EXCLUDE_FROM_ALL)
-endif()
+find_package(GTest REQUIRED)
 
 set(GTEST_EXECUTABLES
     dispatcher_tests
