diff --git a/pdf2svg.c b/pdf2svg.c
index cb5a3ae..8540d78 100644
--- a/pdf2svg.c
+++ b/pdf2svg.c
@@ -67,6 +67,11 @@ int convertPage(PopplerPage *page, const char* svgFilename)
 	cairo_svg_surface_set_document_unit(surface, CAIRO_SVG_UNIT_PT);
 	drawcontext = cairo_create(surface);
 
+	// Draw white background
+	cairo_set_source_rgb(drawcontext, 1.0, 1.0, 1.0); // white
+	cairo_rectangle(drawcontext, 0, 0, width, height);
+	cairo_fill(drawcontext);
+
 	// Render the PDF file into the SVG file
 	poppler_page_render_for_printing(page, drawcontext);
 	cairo_show_page(drawcontext);
