diff --git a/pdf2svg.c b/pdf2svg.c
index cb5a3ae..12c024f 100644
--- a/pdf2svg.c
+++ b/pdf2svg.c
@@ -62,12 +62,15 @@ int convertPage(PopplerPage *page, const char* svgFilename)
 	}
 	poppler_page_get_size (page, &width, &height);
 
+	const double scale = 1.5;
+
 	// Open the SVG file
-	surface = cairo_svg_surface_create(svgFilename, width, height);
+	surface = cairo_svg_surface_create(svgFilename, width * scale, height * scale);
 	cairo_svg_surface_set_document_unit(surface, CAIRO_SVG_UNIT_PT);
 	drawcontext = cairo_create(surface);
 
 	// Render the PDF file into the SVG file
+	cairo_scale(drawcontext, scale, scale);
 	poppler_page_render_for_printing(page, drawcontext);
 	cairo_show_page(drawcontext);
 
