diff -ur aocl-utils-4.1_orig/CMakeLists.txt aocl-utils-4.1/CMakeLists.txt
--- aocl-utils-4.1_orig/CMakeLists.txt	2024-02-15 23:47:46.940884482 +0800
+++ aocl-utils-4.1/CMakeLists.txt	2024-02-15 23:52:04.330823743 +0800
@@ -44,7 +44,6 @@
 SET(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}")
 SET(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}")
 
-SET(CMAKE_INSTALL_LIBDIR "lib")
 
 OPTION(ALCI_DOCS "ENABLE DOCUMENTATION" ON)
 OPTION(ALCI_DOXYGEN_DOC "ENABLE DOXYGEN DOCUMENTATION" OFF)
diff -ur aocl-utils-4.1_orig/lib/CMakeLists.txt aocl-utils-4.1/lib/CMakeLists.txt
--- aocl-utils-4.1_orig/lib/CMakeLists.txt	2024-02-15 23:47:46.940884482 +0800
+++ aocl-utils-4.1/lib/CMakeLists.txt	2024-02-15 23:48:37.210872144 +0800
@@ -32,7 +32,7 @@
   ENDIF()
   # Install .dll into lib directory to have .lib and .dll in same path for shared library
   INSTALL(FILES "${PROJECT_BINARY_DIR}/${CMAKE_BUILD_TYPE}/${CMAKE_SHARED_LIBRARY_PREFIX}${PROJECT_NAME}${AOCL_APPEND_BUILDTYPE_FOR_DLL}.dll"
-          DESTINATION "${CMAKE_INSTALL_PREFIX}/lib")
+		  DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
 ENDIF()
 
 INCLUDE_DIRECTORIES(
