diff --git a/CMake/au_find_gtest.cmake b/CMake/au_find_gtest.cmake
index 36a9369..4683391 100644
--- a/CMake/au_find_gtest.cmake
+++ b/CMake/au_find_gtest.cmake
@@ -1,18 +1,13 @@
-include(FetchContent)
 set(INSTALL_GTEST OFF)
+find_package(GTest REQUIRED)
 include(GoogleTest)
-FetchContent_Declare(gtest
-  GIT_REPOSITORY https://github.com/google/googletest.git
-  GIT_TAG release-1.12.1
-  )
-FetchContent_MakeAvailable(gtest)
 
 #
 # On Windows: Prevent overriding the parent project's compiler/linker settings
 #
 IF(WIN32)
-	target_link_libraries(gmock PUBLIC gtest)
-	target_link_libraries(gmock_main PUBLIC gtest_main)
+	target_link_libraries(gmock PUBLIC GTest::gtest)
+	target_link_libraries(gmock_main PUBLIC GTest::gtest_main)
 ENDIF()
 
 set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
@@ -22,8 +17,8 @@ option(GMOCK_ENABLE_INSTALL "" OFF)
 #FetchContent_MakeAvailable(gtest)
 # You can disable clang-tidy for specific folders in the fetched content project
 if(CMAKE_CXX_CLANG_TIDY)
-    set_target_properties(gtest PROPERTIES CXX_CLANG_TIDY "")
-    set_target_properties(gtest_main PROPERTIES CXX_CLANG_TIDY "")
-    set_target_properties(gmock PROPERTIES CXX_CLANG_TIDY "")
-    set_target_properties(gmock_main PROPERTIES CXX_CLANG_TIDY "")
+    set_target_properties(GTest::gtest PROPERTIES CXX_CLANG_TIDY "")
+    set_target_properties(GTest::gtest_main PROPERTIES CXX_CLANG_TIDY "")
+    set_target_properties(GTest::gmock PROPERTIES CXX_CLANG_TIDY "")
+    set_target_properties(GTest::gmock_main PROPERTIES CXX_CLANG_TIDY "")
 endif()
diff --git a/CMake/au_unit_tests.cmake b/CMake/au_unit_tests.cmake
index 8682acd..c28f416 100644
--- a/CMake/au_unit_tests.cmake
+++ b/CMake/au_unit_tests.cmake
@@ -127,7 +127,7 @@ function(au_cc_test testName)
   )
 
   # Add gtest with main() as dependency
-  target_link_libraries(${_target_name} PRIVATE gmock_main)
+  target_link_libraries(${_target_name} PRIVATE gmock gmock_main gtest)
 
   #
   # Add additional dependencies,
diff --git a/Library/Tests/CMakeLists.txt b/Library/Tests/CMakeLists.txt
index c52afca..ee8fbe4 100644
--- a/Library/Tests/CMakeLists.txt
+++ b/Library/Tests/CMakeLists.txt
@@ -145,15 +145,15 @@ endif()
 #        au::base64
 #)
 
-au_cc_test(CpuidTest
-  SOURCES
-  ${CPUID_TEST_FILES}
-  HEADERS Mock/MockTest.hh QemuTest.hh
-  DEPENDS
-  au::cpuid
-  EXTLIBRARY
-  Python3
-)
+# au_cc_test(CpuidTest
+#   SOURCES
+#   ${CPUID_TEST_FILES}
+#   HEADERS Mock/MockTest.hh QemuTest.hh
+#   DEPENDS
+#   au::cpuid
+#   EXTLIBRARY
+#   Python3
+# )
 ##########
 # Dynamic Loader Tests
 ##########
