From 0d8daaf649352ef0dc8b0c5ed2bca8391a3dab5c Mon Sep 17 00:00:00 2001
From: wangjiezhe <wangjiezhe@gmail.com>
Date: Mon, 17 Feb 2025 13:55:39 +0800
Subject: [PATCH] Change library directory according to CMake build

---
 setup.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 713e8dc8099..48d89b9624a 100644
--- a/setup.py
+++ b/setup.py
@@ -341,7 +341,8 @@ else:
 
 # Constant known variables used throughout this file
 cwd = os.path.dirname(os.path.abspath(__file__))
-lib_path = os.path.join(cwd, "torch", "lib")
+cmake_build_dir = os.environ.get("CMAKE_BUILD_DIR", os.path.join(cwd, "build"))
+lib_path = os.path.join(cmake_build_dir, "lib")
 third_party_path = os.path.join(cwd, "third_party")
 
 # CMAKE: full path to python library
-- 
2.45.3

