From b7295d2621c92f64bbdb1248d7c225348b0032b4 Mon Sep 17 00:00:00 2001
From: wangjiezhe <wangjiezhe@gmail.com>
Date: Mon, 17 Feb 2025 13:44:01 +0800
Subject: [PATCH] Don't build libtorch again

---
 setup.py | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/setup.py b/setup.py
index a6a6db7b033..2b731fdf6d1 100644
--- a/setup.py
+++ b/setup.py
@@ -259,7 +259,7 @@ from setuptools.dist import Distribution
 from tools.build_pytorch_libs import build_pytorch
 from tools.generate_torch_version import get_torch_version
 from tools.setup_helpers.cmake import CMake
-from tools.setup_helpers.env import build_type, IS_DARWIN, IS_LINUX, IS_WINDOWS
+from tools.setup_helpers.env import IS_DARWIN, IS_LINUX, IS_WINDOWS, build_type
 from tools.setup_helpers.generate_linker_script import gen_linker_script
 
 
@@ -483,14 +483,6 @@ def build_deps():
     check_submodules()
     check_pydep("yaml", "pyyaml")
     build_python = not BUILD_LIBTORCH_WHL
-    build_pytorch(
-        version=version,
-        cmake_python_library=cmake_python_library,
-        build_python=build_python,
-        rerun_cmake=RERUN_CMAKE,
-        cmake_only=CMAKE_ONLY,
-        cmake=cmake,
-    )
 
     if CMAKE_ONLY:
         report(
-- 
2.45.3

