From 79b085bdf649cdc7a5d4e74f26adad5c5cc0c5df Mon Sep 17 00:00:00 2001
From: wangjiezhe <wangjiezhe@gmail.com>
Date: Mon, 17 Feb 2025 13:46:35 +0800
Subject: [PATCH] Fix sys.path

---
 setup.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/setup.py b/setup.py
index 2b731fdf6d1..713e8dc8099 100644
--- a/setup.py
+++ b/setup.py
@@ -256,6 +256,10 @@ import setuptools.command.install
 import setuptools.command.sdist
 from setuptools import Extension, find_packages, setup
 from setuptools.dist import Distribution
+
+
+sys.path[:0] = os.getcwd()
+
 from tools.build_pytorch_libs import build_pytorch
 from tools.generate_torch_version import get_torch_version
 from tools.setup_helpers.cmake import CMake
-- 
2.45.3

