diff --git a/caffe2/utils/proto_utils.cc b/caffe2/utils/proto_utils.cc
index 8fc81586f3c..1fa36924e18 100644
--- a/caffe2/utils/proto_utils.cc
+++ b/caffe2/utils/proto_utils.cc
@@ -136,13 +136,7 @@ class IfstreamInputStream : public ::google::protobuf::io::CopyingInputStream {
 } // namespace
 
 C10_EXPORT string ProtoDebugString(const MessageLite& proto) {
-  string serialized = proto.SerializeAsString();
-  for (char& c : serialized) {
-    if (c < 0x20 || c >= 0x7f) {
-      c = '?';
-    }
-  }
-  return serialized;
+  return proto.ShortDebugString();
 }
 
 C10_EXPORT bool ParseProtoFromLargeString(
diff --git a/caffe2/utils/proto_utils.h b/caffe2/utils/proto_utils.h
index a6903425ab4..03dbaef0377 100644
--- a/caffe2/utils/proto_utils.h
+++ b/caffe2/utils/proto_utils.h
@@ -71,8 +71,27 @@ inline bool ParseFromString(const string& spec, MessageLite* proto) {
 
 
 TORCH_API string ProtoDebugString(const MessageLite& proto);
+inline string ProtoDebugString(const MessageLite& proto) {
+  return proto.ShortDebugString();
+}
+
+inline void setTotalBytesLimit(::google::protobuf::io::CodedInputStream& stream, int bytes_limit, int warning_threshold) {
+  #if GOOGLE_PROTOBUF_VERSION >= 3011000
+    // Only take one parameter since protobuf 3.11
+    stream.SetTotalBytesLimit(bytes_limit);
+  #else
+    stream.SetTotalBytesLimit(bytes_limit, warning_threshold);
+  #endif
+}
 
 TORCH_API bool ParseProtoFromLargeString(const string& str, MessageLite* proto);
+inline bool ParseProtoFromLargeString(const string& str, MessageLite* proto) {
+  ::google::protobuf::io::ArrayInputStream input_stream(str.data(), str.size());
+  ::google::protobuf::io::CodedInputStream coded_stream(&input_stream);
+  // Set PlanDef message size limit to 2G.
+  setTotalBytesLimit(coded_stream, 2147483647, 512LL << 20);
+  return proto->ParseFromCodedStream(&coded_stream);
+}
 
 // Text format MessageLite wrappers: these functions do nothing but just
 // allowing things to compile. It will produce a runtime error if you are using
