diff --git a/aten/src/ATen/native/sparse/cuda/cuSPARSELtOps.cpp b/aten/src/ATen/native/sparse/cuda/cuSPARSELtOps.cpp
index c2692441cd7..d1d9c88f64b 100644
--- a/aten/src/ATen/native/sparse/cuda/cuSPARSELtOps.cpp
+++ b/aten/src/ATen/native/sparse/cuda/cuSPARSELtOps.cpp
@@ -125,15 +125,15 @@ at::Tensor _cslt_sparse_mm(
         break;
     case at::ScalarType::Half:
         type = CUDA_R_16F;
-        compute_type = CUSPARSE_COMPUTE_16F;
+        compute_type = CUSPARSE_COMPUTE_32F;
         break;
     case at::ScalarType::BFloat16:
         type = CUDA_R_16BF;
-        compute_type = CUSPARSE_COMPUTE_16F;
+        compute_type = CUSPARSE_COMPUTE_32F;
         break;
     case at::ScalarType::Float:
         type = CUDA_R_32F;
-        compute_type = CUSPARSE_COMPUTE_TF32;
+        compute_type = CUSPARSE_COMPUTE_32F;
         break;
     default:
         TORCH_CHECK(false, "Unsupported dtype for cuSPARSE compressed matrix multiplication.");
