diff --git a/caffe2/CMakeLists.txt b/caffe2/CMakeLists.txt
index 89c31fab113..1cc9576c28b 100644
--- a/caffe2/CMakeLists.txt
+++ b/caffe2/CMakeLists.txt
@@ -1651,6 +1651,9 @@ if(BUILD_SHARED_LIBS)
     target_link_libraries(torch_global_deps ${Caffe2_PUBLIC_CUDA_DEPENDENCY_LIBS})
     target_link_libraries(torch_global_deps torch::cudart torch::nvtoolsext)
   endif()
+  if(USE_OPENMP)
+    target_link_libraries(torch_global_deps OpenMP::OpenMP_CXX)
+  endif()
   install(TARGETS torch_global_deps DESTINATION "${TORCH_INSTALL_LIB_DIR}")
 endif()
 
