diff --git a/caffe2/contrib/opencl/context.cc b/caffe2/contrib/opencl/context.cc
index f0cf9c3353c..c7e01415a5d 100644
--- a/caffe2/contrib/opencl/context.cc
+++ b/caffe2/contrib/opencl/context.cc
@@ -7,7 +7,7 @@
 
 namespace caffe2 {
 
-CAFFE_KNOWN_TYPE(Tensor<OpenCLContext>);
+// CAFFE_KNOWN_TYPE(Tensor<OpenCLContext>);
 
 OpenCLContextSingleton::OpenCLContextSingleton() {
   const auto platform_id = 0;
