diff --git a/gtests/SConscript b/gtests/SConscript
index d963203..5ae8b54 100755
--- a/gtests/SConscript
+++ b/gtests/SConscript
@@ -200,7 +200,7 @@ else:
     liblist.extend(['msvcrt', 'gtest', 'benchmark','mparith32', 'mparith64', 'mpfr', 'mpc'])
 #glibpath = ['gapi/gtest', 'gapi/gbench']
 glibpath = ['gapi/gtest', 'gapi/gbench','libs/mparith/32', 'libs/mparith/64']
-libpath  += [joinpath('#', build_root, x) for x in glibpath]
+libpath  = [joinpath('#', build_root, x) for x in glibpath] + libpath
 
 #if intel svml
 if abi == 'svml':
