diff -ur libflame-4.1_orig/src/map/lapack2flamec/FLA_getrf.c libflame-4.1/src/map/lapack2flamec/FLA_getrf.c
--- libflame-4.1_orig/src/map/lapack2flamec/FLA_getrf.c	2024-02-16 11:47:41.758231053 +0800
+++ libflame-4.1/src/map/lapack2flamec/FLA_getrf.c	2024-02-16 13:46:51.197203430 +0800
@@ -232,6 +232,7 @@
 
 #else /* FLA_ENABLE_SUPERMATRIX */
 
+#define LAPACK_getrf_body_s LAPACK_getrf_body
 #define LAPACK_getrf_body_d LAPACK_getrf_body
 #define LAPACK_getrf_body_z LAPACK_getrf_body
 
Only in libflame-4.1: .vscode
